src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,195 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private void linkLast(T value) {
+        final Node<T> last = tail;
+        final Node<T> newNode = new Node<>(last, value, null);
+        tail = newNode;
+        if (last == null) {
+            head = newNode;
+        } else {
+            last.next = newNode;
+        }
+        size++;
+    }
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndex(index);
+
+        if (index == size) {
+            linkLast(value);
+        } else {
+            linkBefore(value, node(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        checkPositionIndex(size);
+
+        Object[] obj = list.toArray();
+        int length = obj.length;
+
+        if (length == 0) {
+            return;
+        }
+
+        Node<T> prev;
+        prev = tail;
+
+        for (Object object : obj) {
+            T value = (T) object;
+            Node<T> newNode = new Node<>(prev, value, null);
+            if (prev == null) {
+                head = newNode;
+            } else {
+                prev.next = newNode;
+            }
+            prev = newNode;
+        }
+
+        tail = prev;
+        size += length;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkElementByIndex(index);
+        return node(index).element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkElementByIndex(index);
+        Node<T> node = node(index);
+        T oldValue = node.element;
+        node.element = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkElementByIndex(index);
+        return unlink(node(index));
     }
 
     @Override
     public boolean remove(T object) {
+        if (object == null) {
+            for (Node<T> node = head; node != null; node = node.next) {
+                if (node.element == null) {
+                    unlink(node);
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> node = head; node != null; node = node.next) {
+                if (object.equals(node.element)) {
+                    unlink(node);
+                    return true;
+                }
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkElementByIndex(int index) {
+        if (index >= 0 && index < size) {
+            return;
+        }
+        throw new IndexOutOfBoundsException();
```

Comment:
you can do if statement for throw new IndexOutOfBoundsException() and avoid return;